[Lynn Low] iP

lynnlow175 opened the pull request:
# Lippy
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

lippy frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 💯 

And it is **FREE**!

Features:

- [x] Managing todos
- [ ] Managing deadlines 
- [ ] Managing events
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0

Commit:
Level 1

Commit:
Level 2

Commit:
Level 3

Commit:
Level 4

Commit:
A-TextUiTesting

Commit:
Level 5

Commit:
Level 6

Commit:
Level 7

Commit:
Rectified Level 7

Commit:
Level 8

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
Test

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Merge branch 'A-MoreOOP'

Commit:
hotfix: delete wrongly generated files

Commit:
Rename package to duke

Commit:
Merge branch 'add-gradle-support'

Commit:
A-JUnit

lynnlow175 committed:
A-JUnit

lynnlow175 committed:
Merge branch 'A-JUnit'

lynnlow175 committed:
A-JavaDoc

lynnlow175 committed:
Merge branch 'branch-A-JavaDoc'

lynnlow175 committed:
A-CodingStandard

lynnlow175 committed:
Merge branch 'branch-A-CodingStandard'

lynnlow175 committed:
Level 9

lynnlow175 committed:
Merge branch 'branch-Level-9'

ktzy0305 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,242 @@
+     */
+
+    public static LocalDateTime dateFormatter(String dateTime) throws DateTimeParseException {
+            String[] inputs = dateTime.split(" ");
+            DateTimeFormatter formatter;
+            if (inputs.length == 2) {
+                formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+                return LocalDateTime.parse(dateTime, formatter);
+            } else {
+                return null;
+            }
+        }
```
It seems that there are an extra 4 spaces of indentation for lines 192 to 200.

ktzy0305 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,50 @@
+        return taskList;
+    }
+
+    /**
+     * Saves the task-list into the file whenever a change is made
+     * @param taskList an array of tasks
+     */
```
Could add an empty line between the method description and the parameters section.

ktzy0305 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,50 @@
+     */
+
+    public static void save (TaskList taskList) {
+            String filepath = "data/duke.txt";
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(filepath))) {
+                writer.write("");
+                for (int i = 1; i <= taskList.size(); i++) {
+                    taskList.getTask(i).save(filepath);
+                }
+            } catch (IOException e) {
+                System.out.println("IOException");
+            }
+        }
+    }
```
It seems that there are an extra 4 spaces of indentation for lines 40 to 50.

ktzy0305 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,26 @@
+ */
+
+public class Todo extends Task {
+    private String Input;
```
Perhaps rename to `input` since variable names should be in camel case?

ktzy0305 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,169 @@
+     * @param deleted of type String that represents the deleted task
+     * @param taskList an array of tasks
+     */
+    public static void removeTask(String deleted, TaskList taskList) {
+        System.out.println("Noted. I've removed this task:\n" + deleted + "\n"
+                + "Now you have " + taskList.size() + " tasks in the list.");
+    }
```
Good indentation for wrapped lines and for breaking before an operator👍

ktzy0305 commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,43 @@
+
+import java.io.*;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I've noticed a few instances where packaged imports (wildcard imports) are used. Do remember to list out the imported classes explicitly.

ktzy0305 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,94 @@
+ */
+public class Task {
+    private String name;
+    private Boolean done;
```
Maybe add a prefix such as `is` before `done` such that the variable sounds more "boolean like"?

ktzy0305 reviewed (commented):
Generally good adherence to coding standards, with minor issues to fix.

AlyssaPng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,242 @@
+                    Ui.printNotDone(thisTask);
+                }
+            } catch (IndexOutOfBoundsException ex) {
+                Ui.outOfBounds();
```
I like how you have personalised ui messages for each exception.

AlyssaPng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,242 @@
+     * @param isLoading a flag to check if the application is restoring
+     */
+    public void handleInput(String input, TaskList taskList, Boolean isLoading) {
+        if (input.equals("bye")) {
+            isExit = true;
+            return;
+        }
+        if (input.startsWith("mark") || input.startsWith("unmark ")) {
+            this.handleMark(input, taskList, isLoading);
+            return;
+        }
+        if (input.startsWith("todo")) {
+            this.handleTodo(input, taskList, isLoading);
+            return;
+        }
+        if (input.startsWith("deadline")) {
+            this.handleDeadline(input, taskList, isLoading);
+            return;
+        }
+        if (input.startsWith("event")) {
+            this.handleEvent(input, taskList, isLoading);
+            return;
+        }
+        if (input.startsWith("delete")) {
+            this.handleDelete(input, taskList);
+            return;
+        }
+        if (input.equals("list")) {
+            this.handleList(taskList);
+            return;
+        }
```
is there a way to write this in if-else statements?

AlyssaPng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,242 @@
+     * @param taskList an array of tasks
+     */
+
+    public void findMatching(String keyword, TaskList taskList) {
```
does this method naming use nouns? can this be changed to `findMatches`?

AlyssaPng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,55 @@
+     *
+     * @param task
+     */
+    public void add(Task task) {
```
Maybe you can specify what you're adding?

AlyssaPng reviewed (commented):
Looks good!! Your code looks great for the most part, but there are just a few small issues here and there that don't quite meet the coding standards.

lynnlow175 committed:
refactor

lynnlow175 committed:
Level 10

lynnlow175 committed:
Level 10

lynnlow175 committed:
Merge branch 'branch-Level-10'

lynnlow175 committed:
change ui

lynnlow175 committed:
Added Assertion Statements

There exist certain assumptions made about the arguments being
passed into some methods.

It is important that we add assertion to check for possible bugs and
terminate the execution so that it is more safe.

Assertion statements have been used within the method to check for certain
characteristics of the arguments (eg. String input is not null)

lynnlow175 committed:
Merge pull request #1 from lynnlow175/branch-A-Assertions

Added Assertion Statements

lynnlow175 committed:
Improved Code Quality

Long method is written.

This affects code readability.

The long method was shortened by abstracting out a chunk into a new method.

This ensures that the logic flow is still the same as the shortened method
will still call the new method.

lynnlow175 committed:
Merge pull request #2 from lynnlow175/branch-A-CodeQuality

Improved Code Quality

lynnlow175 committed:
Addition of Extension

There is no mechanism to check for duplicate task.

It is important to check for duplicate so that the task list will
correctly reflect the tasks that need to be completed.

A for loop is implemented to check if the new task is the same as any
of the existing tasks and if yes, the new task will not be added to the
task list.

lynnlow175 committed:
A better GUI

lynnlow175 committed:
Merge branch 'A-BetterGui'

lynnlow175 committed:
Ui.png

lynnlow175 committed:
edited Ui.png

lynnlow175 committed:
User Guide

lynnlow175 committed:
fix bug

lynnlow175 committed:
fix bug

lynnlow175 committed:
refactor code

lynnlow175 committed:
A-MoreErrorHandling

lynnlow175 committed:
Merge branch 'A-MoreErrorHandling'

lynnlow175 committed:
Update README.md
